plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -140,26 +141,26 @@ public CacheStatsMBean getMetadataCacheStats()
     }
 
     public TableSnapshot loadSnapshot(SchemaTableName table, Path tableLocation, ConnectorSession session)
-            throws IOException
+            throws IOException, ExecutionException
     {
         String location = tableLocation.toString();
         TableSnapshot cachedSnapshot = tableSnapshots.getIfPresent(location);
         TableSnapshot snapshot;
         FileSystem fileSystem = getFileSystem(tableLocation, table, session);
         if (cachedSnapshot == null) {
-            snapshot = TableSnapshot.load(
-                    table,
-                    fileSystem,
-                    tableLocation,
-                    parquetReaderOptions,
-                    checkpointRowStatisticsWritingEnabled);
-            tableSnapshots.put(location, snapshot);
+            snapshot = tableSnapshots.get(location, () ->
+                    TableSnapshot.load(
+                            table,
+                            fileSystem,
+                            tableLocation,
+                            parquetReaderOptions,
```

Comment:
`parquetReaderOptions` and `checkpointRowStatisticsWritingEnabled` are used only for `TableSnapshot.load` within the `TransactionLogAccess` class.

I recommend extracting the logic of the static method to a class which can be injected into `TransactionLogAccess` . This would give us the opportunity to thoroughly test the fix that you are applying in this PR.

Reply:
Sorry I don't understand this comment. Which static method are you talking about?

Reply:
`io.trino.plugin.deltalake.transactionlog.TableSnapshot#load`

Reply:
Do you mean `io.trino.plugin.deltalake.transactionlog.TableSnapshot.loadSnapshot` ? It is not static this is why I am confused. 
Also if you extract this to another class what about `io.trino.plugin.deltalake.transactionlog.TableSnapshot.getActiveFiles` ? Would you also try to extract it ?

Reply:
I spotted these settings a bit out of place while reading the PR.

Obviously, it is not mandatory to extract this logic for being able to test the race conditions. 
Ideally, this PR should provide appropriate changes to ensure that the class correctly with race conditions.

Reply:
I agree that this kind of tests would be good but I am not sure how to implement them. Running some simple operations hundred of times and waiting for failure/success seems like a flaky test to me. 
I could maybe simulate something with heavy mocking but I have an impression like this kid of tests are not really used here